sse/sse.go

```diff
@@ -209,61 +290,71 @@ func (es *EventSource) dispatch() {
             ch, exists := es.channels[c.channel]
 
             if !exists {
-                ch = make(map[*Client]bool)
+                ch = &Channel{
+                                new(sync.Mutex),
+                                "",
+                                make(map[*Client]bool),
+                                0,
+                            }
                 es.channels[c.channel] = ch
                 log.Info("New channel '%s' created.", c.channel)
             }
 
-            ch[c] = true
+            ch.UpdateExpiration()
+
+            ch.clientsConnected[c] = true
             log.Info("Client '%s' connected to channel '%s'.", c.ip, c.channel)
+            lastMessage := ch.GetLastMessage()
+
+            if (es.UseLastMessage && (len(lastMessage) > 0)) {
```

Comment:
Yes, wee need implement this, but not now. The feature lastMessage is disable by default, and not have impact the actually implementation. Even some user use this feature, the worst scenario occurs when clients receiving redundant messages in connections loss.